### PR TITLE
LRP Data Parallel Support and Fixes

### PR DESCRIPTION
**Pull request by vivekmig**
This adds data parallel (GPU) support to LRP along with relevant tests as well as target and hook removal tests for LRP.

In addition, this fixes the follow issues:
- Multiplying relevance with output was previously done by unsqueezing dim 1, which did not appropriately handle all relevance / input dimensions
- In normalized_relevances for LRP, cases between tensor and tuple were not previously differentiated, causing extraneous output tensors to be returned when attempting to iterate over the first dimension of the tensor (rather than the expected tuple case)
- Deltas were previously computed separately for multiple input tensors in LRP, switched to return single delta tensor

Formatting changes in other files due to upgrading black version

### CODE REVIEW

File: tests/attr/helpers/test_config.py [hunk 1]
```diff
@@ -20,7 +20,9 @@
 from captum.attr._core.layer.layer_gradient_shap import LayerGradientShap
 from captum.attr._core.layer.layer_gradient_x_activation import LayerGradientXActivation
 from captum.attr._core.layer.layer_integrated_gradients import LayerIntegratedGradients
+from captum.attr._core.layer.layer_lrp import LayerLRP
 from captum.attr._core.lime import Lime
+from captum.attr._core.lrp import LRP
```
Comment (NarineK) [inline tests/attr/helpers/test_config.py:25]: Are all configurations here being run by data parallel test ?

**Review by NarineK — APPROVED** [review]
LGTM! Thank you for working on this!

File: captum/attr/_core/layer/layer_lrp.py [hunk 2]
```diff
@@ -41,6 +41,8 @@ def __init__(self, model, layer) -> None:
         """
         LayerAttribution.__init__(self, model, layer)
         LRP.__init__(self, model)
+        if hasattr(self.model, "device_ids"):
```
[the remark this answers is not in the document]
Comment (vivekmig) [reply captum/attr/_core/layer/layer_lrp.py:56]: device_ids is needed to combine the split outputs / gradients in DataParallel models. I think we currently include device_ids as a parameter in any layer attribution method which takes a forward function rather than a model (all except LayerDeepLift). If the DataParallel model is provided itself, then we can just get the device_ids from it; it only needs to be passed when a wrapper forward function is used, which isn't feasible with LayerDeepLift and LRP. I think LayerDeepLift does make use of device_ids internally in the same way through utility methods, but since LRP needs this more frequently, I've just included it as a direct instance variable.

File: tests/attr/test_hook_removal.py [hunk 3]
```diff
@@ -43,6 +44,12 @@ class HookRemovalMode(Enum):
 
 
 class ErrorModule(Module):
+    def __init__(
+        self,
+    ):
+        super().__init__()
+        self.relu = torch.nn.ReLU()
```
[the remark this answers is not in the document]
Comment (vivekmig) [reply tests/attr/test_hook_removal.py:51]: Yes, I think we needed to add a child module for hooks to be set.

File: tests/attr/helpers/test_config.py (hunk 1 above)
Comment (vivekmig) [reply tests/attr/helpers/test_config.py:25]: Yes, each config is run as a data parallel test

File: captum/attr/_core/lrp.py [hunk 4]
```diff
@@ -222,19 +214,12 @@ def compute_convergence_delta(self, attributions, output):
...
         if isinstance(attributions, tuple):
-            stacked_attributions = torch.stack(attributions, dim=1)
-            delta = _attribution_delta(stacked_attributions, output)
+            for attr in attributions:
+                summed_attr = sum(_sum_rows(attr) for attr in attributions)
         else:
-            delta = _attribution_delta(attributions, output)
-        return delta
+            summed_attr = _sum_rows(attributions)
+        return output.flatten() - summed_attr.flatten()
```
[the remark this answers is not in the document]
Comment (vivekmig) [reply captum/attr/_core/lrp.py:257]: I think the difference is that the GradientAttribution implementation performs forward passes on the start and end points; in this case, since we already have the output computed, so this may be more efficient. Maybe we can refactor a bit to share this part in a separate diff?

File: captum/attr/_core/layer/layer_lrp.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (NarineK) [reply captum/attr/_core/layer/layer_lrp.py:56]: sounds good, it looks like only LayerDeepLift doesn't have it and it takes model not forward function.

File: captum/attr/_core/lrp.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (NarineK) [reply captum/attr/_core/lrp.py:259]: Yes, `compute_convergence_delta` does call forward internally. I think it is concise at this point too. We can leave it as is.
